src/Bolt.php

```diff
@@ -195,67 +181,75 @@ private function packProtocolVersions(): string
 
     /**
      * Send INIT message
-     * @version <3
-     * @param string $name should conform to "Name/Version" https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent
-     * @param string $user
-     * @param string $password
-     * @param array $routing routing::Dictionary(address::String)
-     <pre>null - the server should not carry out routing
-     [] - the server should carry out routing
-     ['address' => 'ip:port'] - the server should carry out routing according to the given routing context</pre>
-     * @param array $metadata Server success response metadata
-     * @return bool
+     *
+     * @note The usage of $user, $password, $routing and $metadata is deprecated. Please use helpers\Auth to generate an authentication strategy as an array.
+     *
+     * @param array|string $nameOrExtra You can use helpers\Auth to generate required array or use the deprecated approach to fill in $name, $user and $password
+     * @param string|null $user
+     * @param string|null $password
+     * @param array|null $routing
+     * @param array $metadata
+     *
+     * @return array
      * @throws Exception
+     * @version <3
      */
-    public function init(string $name, string $user, string $password, array $routing = null, array &$metadata = []): bool
+    public function init($nameOrExtra, string $user = null, string $password = null, array $routing = null, array $metadata = []): array
     {
-        if (!$this->connection->connect())
```

Comment:
If this API change is fine by you @stefanak-michal, we can merge it as far as I am concerned. I don't have a good way to test it with OIDC, so it might be wise if you run your tests locally again

Reply:
updated to follow original logic and tested